frontend/src/features/documents/documentUpload/DocumentUploadView.tsx

```diff
@@ -0,0 +1,206 @@
+import { Button } from 'components/common/buttons/Button';
+import { Input, Select, SelectOption } from 'components/common/form';
+import { Scrollable } from 'components/common/Scrollable/Scrollable';
+import TooltipIcon from 'components/common/TooltipIcon';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import * as API from 'constants/API';
+import { SectionField } from 'features/mapSideBar/tabs/SectionField';
+import { Formik } from 'formik';
+import useLookupCodeHelpers from 'hooks/useLookupCodeHelpers';
+import { Api_DocumentType, Api_UploadRequest } from 'models/api/Document';
+import { Api_Storage_DocumentTypeMetadataType } from 'models/api/DocumentStorage';
+import { ChangeEvent, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+
+interface DocumentUploadForm {
+  documentTypeId: string;
+  documentStatusCode: string;
+}
+
+interface IDocumentUploadViewProps {
+  documentTypes: Api_DocumentType[];
+  isLoading: boolean;
+  mayanMetadata: Api_Storage_DocumentTypeMetadataType[];
```

Comment:
nit: please consider renaming this prop to "storageMetadata"

Reply:
Also, it is usually helpful (in tests etc) to be able to import the interface type - please consider exporting the interfaces as well here